File: content/atlas/source/includes/fts/field-types/embeddedDocuments/stored-source-relative-path.py
from pymongo import MongoClient
from pymongo.operations import SearchIndexModel

# Connect to your Atlas deployment
uri = "<connection-string>"
client = MongoClient(uri)

database = client["sample_training"]
collection = database["companies"]

# Create the Atlas Search index definition for the embeddedDocuments field
search_index_model = SearchIndexModel(
    definition={
        "mappings": { 
          "dynamic": false,
            "fields": { 
            "funding_rounds": { 
              "type": "embeddedDocuments",
              "dynamic": true,
              "storedSource": {
                "include": ["round_code", "raised_currency_code", "raised_amount"] 
              }
            } 
          }
        }
    },
    name="default",
)

# Create the index
result = collection.create_search_index(model=search_index_model)
print(f"New index name: {result}")

File: content/atlas/source/includes/fts/field-types/embeddedDocuments/stored-source-relative-path.cpp
#include <iostream>

#include <bsoncxx/builder/basic/document.hpp>
#include <bsoncxx/json.hpp>
#include <mongocxx/client.hpp>
#include <mongocxx/instance.hpp>
#include <mongocxx/uri.hpp>

using bsoncxx::builder::basic::kvp;
using bsoncxx::builder::basic::make_document;

int main() {
    // Initialize the MongoDB C++ driver
    mongocxx::instance instance{};

    // Connect to your MongoDB deployment
    mongocxx::uri uri("<connection-string>");
    mongocxx::client client(uri);

    auto db = client["sample_training"];
    auto collection = db["companies"];
    auto siv = collection.search_indexes();

    // Create the Atlas Search index definition for the embeddedDocuments field
    auto name = "default";
    auto definition = make_document(
        kvp("mappings", make_document(
            kvp("dynamic", false),
            kvp("fields", make_document(
                kvp("funding_rounds", make_document(
                    kvp("type", "embeddedDocuments"),
                    kvp("dynamic", true),
                    kvp("storedSource", make_document(
                        kvp("include", bsoncxx::builder::basic::array{
                            "round_code", "raised_currency_code", "raised_amount"
                        })
                    ))
                ))
            ))
        ))
    );
    auto model = mongocxx::search_index_model(name, definition.view());

    try {
        // Execute the command to create the index
        auto result = siv.create_one(model);
        std::cout << "New index name: " << result << std::endl;
    } catch (const std::exception& e) {
        std::cerr << "Error creating Atlas Search index: " << e.what() << std::endl;
    }
}